.github/workflows/upload_new_tag_to_pypi.yml

```diff
@@ -0,0 +1,58 @@
+name: Publish Python 🐍 distributions 📦 to PyPI and TestPyPI
+
+on: push

```

Comment:
My understanding is that this workflow is enabled on *all* push events (not only to main branch and not just new tags) because you want to check the build process for all pull requests too?

Reply:
That's correct.

Reply:
It's a nice aspect of this workflow.  Too often, a package will pass all the CI but fail in the final packaging steps _before_ upload is attempted.  The routine checks (from `python build` and `twine check`) are appreciated.